umad/src/main/java/com/pkukielka/AccessMonitorRewriter.java

```diff
@@ -144,7 +141,23 @@ private boolean isSafeMethod(ConstPool pool, int index) {
         return safeMethods.contains(fqn);
     }
 
-    private boolean isSafe(CtMethod editableMethod, String dottedName) {
+    private CtMethod getMethodIfExist(ConstPool constPool, int methodIndex)  {
+        try {
+            String className = constPool.getMethodrefClassName(methodIndex);
+            String methodName = constPool.getMethodrefName(methodIndex);
+            if (methodName.endsWith("$")) methodName = methodName.substring(0, methodName.length() - 1);
```

Comment:
Why do we need to drop `$` at the end? It looks buggy to me (I've got a feeling that we may check same method twice that way)

Reply:
I was looking into this extensively in the debugger since it was bugging me as well.
TBH i do not know why that `$` is there but it don't want to disappear ^^

Reply:
Let's merge it and I will take a look